src/sys.zig

```diff
@@ -281,7 +281,7 @@ pub fn getErrno(rc: anytype) bun.C.E {
         return bun.C.E.UNKNOWN;
     }
 
-    if (comptime use_libc) return std.os.errno(rc);
+    if (comptime Environment.isMac) return std.os.errno(rc);
```

Comment:
Note that before this patch, Mac was also falling under the `@TypeOf` check below, which means it was calling `std.os.linux.getErrno` to properly handle `usize` rc.

Does `std.os.errno` have a different implementation on Mac than on Linux?  If not,and you pass an unsigned integer, it might crash because of the `integer cast truncated bits` assertion (https://github.com/oven-sh/bun/pull/6706#issue-1961666697). 

Note that all of the linked issues are Linux/arm64 issues, and only happen on Mac because Docker on Apple silicon runs under  a Linux/arm64 VM. So in the best scenario this condition is not doing anything, in the worst case it is moving the bug to Mac.